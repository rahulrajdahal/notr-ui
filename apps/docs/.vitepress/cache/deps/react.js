import {
  require_react
} from "./chunk-UUF3TEGA.js";
import "./chunk-5WRI5ZAA.js";
export default require_react();
//# sourceMappingURL=react.js.map
