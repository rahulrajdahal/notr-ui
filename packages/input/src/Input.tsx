import { css } from "@emotion/react";
import { colors } from "@rahulrajdahal/notr-core";
import React from "react";

const baseStyles = css`
  display: flex;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
`;
const labelBaseStyles = css`
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
`;
const inputBaseStyles = css`
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;

  &:focus {
    border-color: var(--color-primary-600);
  }
`;
const errorBaseStyles = css`
  font-size: 0.875rem;
  color: var(--color-red-600);
  font-style: italic;
`;
const errorInputStyle = css`
  border-color: var(--color-red-600);
`;

export interface InputProps
  extends React.FieldsetHTMLAttributes<HTMLFieldSetElement> {
  /** Props for the input label component */
  labelProps?: React.ComponentProps<"label">;
  /** Props for the input component */
  inputProps?: React.ComponentProps<"input">;
  /** Label of the input*/
  label: string;
  /** Collection of input errors*/
  errors?: string[];
}

const Input = React.forwardRef<HTMLFieldSetElement, InputProps>(
  ({ label, errors, labelProps, inputProps, ...props }, ref) => {
    return (
      <fieldset
        ref={ref}
        style={
          {
            "--color-primary-600": colors.primary[600],
            "--color-gray-400": colors.gray[400],
            "--color-gray-600": colors.gray[600],
            "--color-red-600": colors.red[600],
          } as React.CSSProperties
        }
        css={baseStyles}
        {...props}
      >
        <label css={labelBaseStyles} htmlFor={inputProps?.name} {...labelProps}>
          {label}
        </label>
        <input
          css={css`
            ${inputBaseStyles} ${errors && errors.length > 0
              ? errorInputStyle
              : ""}
          `}
          {...inputProps}
        />
        {errors &&
          errors.length > 0 &&
          errors.map((error) => (
            <span key={error} css={errorBaseStyles}>
              {error}
            </span>
          ))}
      </fieldset>
    );
  }
);

Input.displayName = "Input";

export default Input;
