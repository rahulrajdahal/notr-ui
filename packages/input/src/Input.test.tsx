import { createSerializer } from "@emotion/jest";
import { colors } from "@rahulrajdahal/notr-core";
import { render, screen } from "@testing-library/react";
import Input from "./Input";

describe("Input Component", () => {
  it("renders with default props and styles", async () => {
    render(<Input data-testid="notr-input" label="Notr Input" />);
    let input = screen.getByTestId("notr-input");
    expect(input).toBeInTheDocument();
    expect(input).toHaveStyle(`--color-primary-600: ${colors.primary[600]}`);
    expect(input).toHaveStyle(`--color-gray-400: ${colors.gray[400]}`);
    expect(input).toHaveStyle(`--color-gray-600: ${colors.gray[600]}`);
    expect(input).toHaveStyle(`--color-red-600: ${colors.red[600]}`);
    expect(input).toHaveStyle("display: flex");
    expect(input).toHaveStyle("flex-direction: column");
    expect(input).toHaveStyle("gap: 0.37rem");

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    input = screen.getByText("Notr Input");
    expect(input).toHaveStyle("font-size: 1rem");
    expect(input).toHaveStyle("font-weight: 500");
    expect(input).toHaveStyle("line-height: 1.25rem");
    expect(input).toHaveStyle("opacity: 0.77");
    expect(input).toHaveStyle("color: var(--color-gray-300)");

    const placeholder = "Notr Input Placeholder";
    render(<Input label="Notr Input" inputProps={{ placeholder }} />);
    input = screen.getByPlaceholderText(placeholder);
    expect(input).toHaveStyle("height: 3.75rem");
    expect(input).toHaveStyle("border-radius: 0.8125rem");
    expect(input).toHaveStyle("border: 1px solid var(--color-gray-600)");
    expect(input).toHaveStyle("padding: 0 0.5rem");
    expect(input).toHaveStyle("outline: none");

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();
  });

  it("renders with errors props and styles", async () => {
    const errors = ["Invalid Input"];
    render(
      <Input data-testid="notr-input" label="Notr Input" errors={errors} />
    );
    let input = screen.getByTestId("notr-input");
    expect(input).toBeInTheDocument();

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    input = screen.getByText("Invalid Input");
    expect(input).toHaveStyle("font-size: 0.875rem");
    expect(input).toHaveStyle("color: var(--color-red-600)");
    expect(input).toHaveStyle("font-style: italic");
  });

  it("renders labelProps and styles", async () => {
    render(
      <Input
        data-testid="notr-input"
        label="Notr Input"
        labelProps={{ title: "Notr Input" }}
      />
    );
    let input = screen.getByTestId("notr-input");
    expect(input).toBeInTheDocument();

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    input = screen.getByText("Notr Input");
    expect(input).toHaveProperty("title", "Notr Input");
    expect(input).toHaveStyle("font-size: 1rem");
    expect(input).toHaveStyle("font-weight: 500");
    expect(input).toHaveStyle("line-height: 1.25rem");
    expect(input).toHaveStyle("opacity: 0.77");
    expect(input).toHaveStyle("color: var(--color-gray-300)");
  });

  it("renders inputProps and styles", async () => {
    const placeholder = "Notr Input";

    render(
      <Input
        data-testid="notr-input"
        label="Notr Input"
        inputProps={{ name: "notr", placeholder }}
      />
    );
    let input = screen.getByTestId("notr-input");
    expect(input).toBeInTheDocument();

    expect.addSnapshotSerializer(createSerializer());
    expect(input).toMatchSnapshot();

    input = screen.getByPlaceholderText("Notr Input");
    expect(input).toHaveProperty("name", "notr");
    expect(input).toHaveProperty("placeholder", placeholder);
    expect(input).toHaveStyle("height: 3.75rem");
    expect(input).toHaveStyle("border-radius: 0.8125rem");
    expect(input).toHaveStyle("border: 1px solid var(--color-gray-600)");
    expect(input).toHaveStyle("padding: 0 0.5rem");
    expect(input).toHaveStyle("outline: none");
  });
});
