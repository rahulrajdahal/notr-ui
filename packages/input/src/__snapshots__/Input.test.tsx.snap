// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Component > renders inputProps and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

<fieldset
  class="emotion-0"
  data-testid="notr-input"
  style="--color-primary-600: #3563de; --color-gray-400: #2a292e; --color-gray-600: #cdcdcd; --color-red-600: #c51d1db6;"
>
  <label
    class="emotion-1"
    for="notr"
  >
    Notr Input
  </label>
  <input
    class="emotion-2"
    name="notr"
    placeholder="Notr Input"
  />
</fieldset>
`;

exports[`Input Component > renders labelProps and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

<fieldset
  class="emotion-0"
  data-testid="notr-input"
  style="--color-primary-600: #3563de; --color-gray-400: #2a292e; --color-gray-600: #cdcdcd; --color-red-600: #c51d1db6;"
>
  <label
    class="emotion-1"
    title="Notr Input"
  >
    Notr Input
  </label>
  <input
    class="emotion-2"
  />
</fieldset>
`;

exports[`Input Component > renders with default props and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

<fieldset
  class="emotion-0"
  data-testid="notr-input"
  style="--color-primary-600: #3563de; --color-gray-400: #2a292e; --color-gray-600: #cdcdcd; --color-red-600: #c51d1db6;"
>
  <label
    class="emotion-1"
  >
    Notr Input
  </label>
  <input
    class="emotion-2"
  />
</fieldset>
`;

exports[`Input Component > renders with default props and styles 2`] = `
.emotion-0 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-0:focus {
  border-color: var(--color-primary-600);
}

.emotion-0 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
}

.emotion-0:focus {
  border-color: var(--color-primary-600);
}

<input
  class="emotion-0"
  placeholder="Notr Input Placeholder"
/>
`;

exports[`Input Component > renders with errors props and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
  border-color: var(--color-red-600);
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-3 {
  font-size: 0.875rem;
  color: var(--color-red-600);
  font-style: italic;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
  border-color: var(--color-red-600);
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-3 {
  font-size: 0.875rem;
  color: var(--color-red-600);
  font-style: italic;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.37rem;
  border: none;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.25rem;
  opacity: 0.77;
  color: var(--color-gray-300);
}

.emotion-2 {
  height: 3.75rem;
  border-radius: 0.8125rem;
  border: 1px solid var(--color-gray-600);
  padding: 0 0.5rem;
  outline: none;
  border-color: var(--color-red-600);
}

.emotion-2:focus {
  border-color: var(--color-primary-600);
}

.emotion-3 {
  font-size: 0.875rem;
  color: var(--color-red-600);
  font-style: italic;
}

<fieldset
  class="emotion-0"
  data-testid="notr-input"
  style="--color-primary-600: #3563de; --color-gray-400: #2a292e; --color-gray-600: #cdcdcd; --color-red-600: #c51d1db6;"
>
  <label
    class="emotion-1"
  >
    Notr Input
  </label>
  <input
    class="emotion-2"
  />
  <span
    class="emotion-3"
  >
    Invalid Input
  </span>
</fieldset>
`;
